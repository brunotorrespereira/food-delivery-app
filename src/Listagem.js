

import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

export default function Listagem({ data }) {
  return (
    <View style={styles.container}>
      <Text style={styles.texto}>Nome:{data.nome}</Text>
      <Text style={styles.texto}>SobreNome:{data.sobreNome}</Text>
      <Text style={styles.texto}>Cargo:{data.cargo}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#FFFFED",
    padding: 10,
    borderBottomWidth: 1,
    borderColor: '#ccc'
  },
  texto: {
    margin:5,
    fontSize: 20,
    color:'#000'
  }
});


