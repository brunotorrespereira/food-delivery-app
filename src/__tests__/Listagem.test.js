
import React from 'react';
import Listagem from '../../src/Listagem';
import renderer from 'react-test-renderer';

describe('Listagem Component', () => {
  it('deve renderizar corretamente os dados fornecidos', () => {
    const data = {
      cargo: 'Desenvolvedor',
      nome: 'Bruno',
      sobreNome: 'Pereira',
    };
    const component = renderer.create(<Listagem data={data} />);
    const tree = component.toJSON();
    
    // Verifica se cada dado está presente na renderização
    expect(tree).toMatchSnapshot();
    expect(tree.children[0].children[0]).toBe(data.cargo);
    expect(tree.children[1].children[0]).toBe(data.nome);
    expect(tree.children[2].children[0]).toBe(data.sobreNome);
  });
});
