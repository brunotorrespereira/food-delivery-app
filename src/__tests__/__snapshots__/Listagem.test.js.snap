// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listagem Component deve renderizar corretamente os dados fornecidos 1`] = `
<View
  style={
    {
      "borderBottomWidth": 1,
      "borderColor": "#ccc",
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 16,
      }
    }
  >
    Desenvolvedor
  </Text>
  <Text
    style={
      {
        "fontSize": 16,
      }
    }
  >
    Bruno
  </Text>
  <Text
    style={
      {
        "fontSize": 16,
      }
    }
  >
    Pereira
  </Text>
</View>
`;
